scripts/migrations/2024_08_validate_and_sync_crm_existing_orgs.pl

```diff
@@ -0,0 +1,627 @@
+#!/usr/bin/perl -w
+
+# This file is part of Product Opener.
+#
+# Product Opener
+# Copyright (C) 2011-2023 Association Open Food Facts
+# Contact: contact@openfoodfacts.org
+# Address: 21 rue des Iles, 94100 Saint-Maur des Fossés, France
+#
+# Product Opener is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use ProductOpener::PerlStandards;
+use Modern::Perl '2017';
+use utf8;
+
+use ProductOpener::Users qw( $User_id );
+use ProductOpener::Orgs qw( list_org_ids retrieve_org store_org );
+use Encode;
+
+binmode(STDOUT, ":encoding(UTF-8)");
+binmode(STDERR, ":encoding(UTF-8)");
+
+# This file is used to:
+# Set the validation status of existing orgs to 'accepted' for the ones in the list,
+# and sync with the CRM.
+# Reject all the others and are sent a rejection email notification.
+# Set the org name if it is missing.
+
+# Set Manon as the salesperson for the orgs
+$User_id = 'manoncorneille';
+
+use Data::Dumper;
+my %accepted = accepted_orgs();
+
+foreach my $org_id (list_org_ids()) {
+	$org_id = decode utf8 => $org_id;
+	my $org_ref = retrieve_org($org_id);
+
+	my $org_name = $org_ref->{name};
+	if (not defined $org_name) {
+		$org_ref->{name} = $org_id =~ s/-/ /gr;
+	}
+
+	my $org_is_valid = exists $accepted{$org_id};
+	if ($org_is_valid) {
+		$org_ref->{valid_org} = 'accepted';
+	}
+	else {
+		$org_ref->{valid_org} = 'rejected';
+	}
+
+	store_org($org_ref);
+	if ($org_is_valid) {
+		print "$org_id\n";
+	}
+}
+
+sub accepted_orgs {
```

Comment:
could have been in a separate CSV @4nt0ineB !

Reply:
yeah, ok might have been lazy on this x). I should probably use a csv and use it as a check point